tests/test_strtree.py

```diff
@@ -8,21 +8,38 @@
 
 from shapely.errors import ShapelyDeprecationWarning
 from shapely.geometry import Point, Polygon
+from shapely.geos import geos_version
 from shapely import strtree
 from shapely.strtree import STRtree
 
 from .conftest import requires_geos_342
 
 
 @requires_geos_342
-def test_query():
-    points = [Point(i, i) for i in range(10)]
+@pytest.mark.parametrize("geoms", [[Point(i, i) for i in range(10)]])
+@pytest.mark.parametrize(
+    "query_geom,num_results",
+    [(Point(2, 2).buffer(0.99), 1), (Point(2, 2).buffer(1.0), 3)],
+)
+def test_query(geoms, query_geom, num_results):
     with pytest.warns(ShapelyDeprecationWarning):
-        tree = STRtree(points)
-    results = tree.query(Point(2, 2).buffer(0.99))
-    assert len(results) == 1
-    results = tree.query(Point(2, 2).buffer(1.0))
-    assert len(results) == 3
+        tree = STRtree(geoms)
+    results = tree.query(query_geom)
+    assert len(results) == num_results
+
+
+@requires_geos_342
+@pytest.mark.parametrize("geoms", [[Point(i, i) for i in range(10)]])
+@pytest.mark.parametrize(
+    "query_geom,expected",
+    [(Point(2, 2).buffer(0.99), [2]), (Point(2, 2).buffer(1.0), [1, 2, 3])],
+)
+def test_query_enumeration_idx(geoms, query_geom, expected):
+    """Store enumeration idx"""
+    with pytest.warns(ShapelyDeprecationWarning):
+        tree = STRtree((g, i) for i, g in enumerate(geoms))
```

Comment:
@sgillies based on your last comments (and the updated top post which mentions *"These sequences are separate, not zipped together"* and your comment at https://github.com/Toblerity/Shapely/pull/1064#issuecomment-813720602) that the idea was the only support geometries and items as separate sequences (where the items is optional), and not in the combined (zipped) form.  
But this test above is testing that constructing the tree from a sequence of tuples actually works?

Reply:
I opened https://github.com/Toblerity/Shapely/pull/1172 to potentially address this

Reply:
Oh my, you're right that this test shouldn't pass! :facepalm: